src/tests/measurements/new/test_signal_processor.py

```diff
@@ -0,0 +1,48 @@
+from unittest import TestCase
+
+from mock import patch
+from numpy import array_equal
+from numpy.ma import array
+from qilib.data_set import DataSet, DataArray
+
+from qtt.measurements.new.interfaces.signal_processor_interface import SignalProcessorInterface
+from qtt.measurements.new.signal_processor import SignalProcessor
+
+
+class TestSignalProcessor(TestCase):
+    def test_run_processes(self):
+        with patch('qtt.measurements.new.interfaces.signal_processor_interface.SignalProcessorInterface') as spi:
+            class DummySignalProcessor(spi):
+                def run_process(self, signal_data: DataSet) -> DataSet:
+                    return signal_data
+
+        signal_processor = SignalProcessor()
+        signal_processor.add_signal_processor(DummySignalProcessor())
+
+        data_set = DataSet()
+        signal_processor.run_processes(data_set)
+        spi.run_process.assert_called_once()
+        spi.run_process.assert_called_with(data_set)
+
+    def test_add_signal_processor(self):
+        class DummySignalProcessor(SignalProcessorInterface):
+            def run_process(self, signal_data: DataSet) -> DataSet:
+                return signal_data
+
+        signal_processor = SignalProcessor()
+        signal_processor.add_signal_processor(DummySignalProcessor())
+        self.assertEqual(len(signal_processor._signal_processors), 1)
+        self.assertIsInstance(signal_processor._signal_processors[0], DummySignalProcessor)
+
+    def test_run_process_single_signal_processor(self):
```

Comment:
I'd expect a test-case for at least running it with two processors as well, and check that the final result is actually processor1(processor2(input). And perhaps running with 0 as well, just to cover the boundary cases. What is the expected behaviour of running with 0?

Reply:
Good one. Done